traits/trait_list_object.py

```diff
@@ -9,426 +9,473 @@
 #  Thanks for using Enthought open source!
 
 import copy
+import operator
 from weakref import ref
 
 from .trait_base import class_of, Undefined
 from .trait_errors import TraitError
 
 
-class TraitListEvent(object):
-    """ An object reporting in-place changes to a traits list.
+def adapt_trait_validator(trait_validator):
+    """ Adapt a trait validator to work as a trait list validator.
     """
 
-    def __init__(self, index=0, removed=None, added=None):
-        """
-        Parameters
-        ----------
-        index : int
-            The location of the first change in the list.
-        added : list
-            The list of values added to the list.
-        removed : list
-            The list of values removed from the list.
-        """
-        self.index = index
+    def validator(trait_list, index, removed, value):
+        try:
+            if isinstance(index, slice):
+                return [
+                    trait_validator(trait_list, "items", item)
+                    for item in value
+                ]
+            else:
+                return trait_validator(trait_list, "items", value)
+        except TraitError as excp:
+            excp.set_prefix("Each element of the")
+            raise excp
 
-        if removed is None:
-            removed = []
-        self.removed = removed
+    return validator
 
-        if added is None:
-            added = []
-        self.added = added
 
+class TraitList(list):
```

Comment:
We still have a `TraitList(TraitHandler)` class, which could be potentially confusing.

Reply:
I know.  I couldn't think of a better name (Naming Things Is Hard) and my understanding is that `TraitList` is eventually going away.  Happy to take suggestions for something better.

Reply:
Not valid any more. (`TraitList` is gone)